From 47f1ace8dd7683d3ecb73976d8f5855285274301 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Sun, 17 Mar 2013 22:41:36 +0100
Subject: [PATCH 0988/1083] drm/rcar-du: Prepare/unprepare clock

Use clk_prepare_enable() and clk_disable_unprepare() to prepare/enable
and disable/unprepare the DU clock.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit 9fc93f05f93d71eee16fe2e0887a96ebe3dd6651)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit 0c895ae31c6e847fd4354da64a46784a1f779969)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_crtc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
index 31ca55e..aa7da6e 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
@@ -233,7 +233,7 @@ static void rcar_du_crtc_start(struct rcar_du_crtc *rcrtc)
 		return;
 
 	/* Enable clocks before accessing the hardware. */
-	clk_enable(rcdu->clock);
+	clk_prepare_enable(rcdu->clock);
 
 	/* Enable extended features */
 	rcar_du_write(rcdu, DEFR, DEFR_CODE | DEFR_DEFE);
@@ -297,7 +297,7 @@ static void rcar_du_crtc_stop(struct rcar_du_crtc *rcrtc)
 
 	rcar_du_start_stop(rcdu, false);
 
-	clk_disable(rcdu->clock);
+	clk_disable_unprepare(rcdu->clock);
 
 	rcrtc->started = false;
 }
-- 
1.8.3.2

